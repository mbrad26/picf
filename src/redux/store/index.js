import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware } from  'redux';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  {},
  applyMiddleware(saga, logger)
);

saga.run(rootSaga);

export default store;