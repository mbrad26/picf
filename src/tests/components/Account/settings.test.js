import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react'

import store from '../../../redux/store';
import Settings from '../../../components/Account/settings';

describe('Settings', () => {
  it('renders a snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Settings />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
