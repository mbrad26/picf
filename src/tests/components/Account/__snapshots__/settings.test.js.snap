// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders a snapshot 1`] = `
<div
  class="settings-container"
>
  <h3>
    Account settings
  </h3>
  <hr />
  <h5>
    Password reset
  </h5>
  <form>
    <input
      name="email"
      placeholder="Email"
      type="email"
      value=""
    />
    <button
      disabled=""
      type="submit"
    >
      Send password reset link
    </button>
  </form>
  <h5>
    Password change
  </h5>
  <form>
    <input
      name="passwordOne"
      placeholder="New password"
      type="text"
      value=""
    />
    <input
      name="passwordTwo"
      placeholder="Confirm new password"
      type="password"
      value=""
    />
    <button
      disabled=""
      type="submit"
    >
      Change password
    </button>
  </form>
</div>
`;
